### PR TITLE
CrayPE fixes for the slepc package

### PR DESCRIPTION
**Pull request by lukebroskop**
Use the gold linker for CCE (this may change later)

### CODE REVIEW

File: var/spack/repos/builtin/packages/slepc/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/slepc/package.py]:
```suggestion
        if self.spec.satisfies('%cce'):
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]

**Comment by tldahlgren** [comment]
LGTM.

@joseeroman @balay Did you want to comment on this PR?

**Review by tldahlgren — APPROVED** [review]

**Comment by tldahlgren** [comment]
Approved but would like to wait for comment(s) from one or both maintainers before merging.

**Comment by joseeroman** [comment]
Why is the gold linker necessary? Does the default linker fail? If this is the case, shouldn't it be fixed upstream in SLEPc?

**Comment by lukebroskop** [comment]
> Why is the gold linker necessary? Does the default linker fail? If this is the case, shouldn't it be fixed upstream in SLEPc?

Yes, the default lld linker failed.

**Comment by tldahlgren** [comment]
> > Why is the gold linker necessary? Does the default linker fail? If this is the case, shouldn't it be fixed upstream in SLEPc?
> 
> Yes, the default lld linker failed.

@joseeroman So are you okay with this PR being merged until SLEPc can resolve this issue (assuming someone reported it)?  Once it's fixed the package can be updated accordingly.

**Comment by joseeroman** [comment]
Ok, go ahead and merge it. I cannot fix this in SLEPc because I cannot reproduce the issue, don't have access to a Cray machine. On the other hand, I don't understand why this issue appears in SLEPc and not in PETSc, since in SLEPc we use the same linker as PETSc.

**Pull request merged**
